src/edu/csus/ecs/pc2/ui/SubmitSampleRunsPane.java

```diff
@@ -0,0 +1,880 @@
+	
+	private JRadioButton submitSelectedJudgingTypeRadioButton = new JRadioButton("Submit Selected");
+	
+    private JCheckBox checkBoxProblems = null;
+
+    private JRadioButton selecteAllProblemsRadioButton = null;
+    private JRadioButton submitAllProblemsRadioButton;
+
+    private JRadioButton submitSelectedProblems = null;
+    private JRadioButton submitSelectedProblemsRadioButton;
+
+    private JLabel selectedProblemsLabel = null;
+
+    private JButton selectProblemsButton = null;
+
+    private JCheckBox checkBoxLanguage = null;
+
+    private JRadioButton selecteAllLanguages = null;
+    private JRadioButton submitAllLanguagesRadioButton;
+
+    private JRadioButton submitSelectedLanguages = null;
+    private JRadioButton submitSelectedLanguagesRadioButton;
+
+    private JLabel selectedLanguagesLabel = null;
+
+    private JButton selectLanguageButton = null;
+
+
+    private Log log;
+
+    private boolean debug22Flag = false;
+
+    // TODO On Admin update of Languages or Update of Problems - clear selected index arrays
+
+	public SubmitSampleRunsPane() {
+		super();
+		setLayout(new BorderLayout(0, 0));
+
+		JPanel centerPane = new JPanel();
+		centerPane.setLayout(new BorderLayout());
+		add(centerPane, BorderLayout.CENTER);
+
+		JSplitPane splitPane = new JSplitPane();
+		splitPane.setOrientation(JSplitPane.VERTICAL_SPLIT);
+		centerPane.add(splitPane, BorderLayout.CENTER);
+
+		JPanel controlsPane = new JPanel();
+		controlsPane.setPreferredSize(new Dimension(400, 400));
+
+		controlsPane.setLayout(null);
+
+		JLabel cdpConfigDirLabel = new JLabel("CDP config dir");
+		cdpConfigDirLabel.setToolTipText("CDP Location for sample source files");
+		cdpConfigDirLabel.setHorizontalAlignment(SwingConstants.RIGHT);
+		cdpConfigDirLabel.setBounds(10, 60, 86, 14);
+		controlsPane.add(cdpConfigDirLabel);
+
+		cdpTextField = new JTextField();
+		cdpTextField.setFont(new Font("Tahoma", Font.PLAIN, 12));
+		cdpTextField.setBounds(113, 54, 418, 27);
+		controlsPane.add(cdpTextField);
+		cdpTextField.setColumns(10);
+		
+        cdpTextField.addKeyListener(new java.awt.event.KeyAdapter() {
+            public void keyPressed(java.awt.event.KeyEvent e) {
+                if (e.getKeyCode() == java.awt.event.KeyEvent.VK_ENTER) {
+                    if (new File(cdpTextField.getText()).isDirectory()) {
+                        updateCDPDirAndFields(cdpTextField.getText());
+                    }
+                }
+
+            }
+        });
+
+		messageLabel = new JLabel("message label");
+		messageLabel.setFont(new Font("Tahoma", Font.PLAIN, 14));
+		messageLabel.setForeground(Color.RED);
+		messageLabel.setHorizontalAlignment(SwingConstants.CENTER);
+		messageLabel.setBounds(10, 11, 691, 32);
+		controlsPane.add(messageLabel);
+
+		checkBoxSubmitYesSamples = new JCheckBox("Submit Yes Samples");
+		checkBoxSubmitYesSamples.setSelected(true);
+		checkBoxSubmitYesSamples.setToolTipText("Only submit AC sample source");
```

Comment:
This tooltip wording seems inconsistent with the intent of the checkbox.  The checkbox itself appears (from what I can tell) to control whether or not the "Yes" samples are to be submitted.  However, the ToolTip describes a different condition: that ONLY the Yes samples are to be submitted.  The two items (the checkbox and the tooltip) need to have agreement on what they are intending to represent.

Reply:
While it's not _critical_, it's much more important than a lot of the comments/suggestions I've made.  I would consider this an approval showstopper.